[DOCS-14568] GA NGINX Single-Step Instrumentation (#37452)
DataDog/documentation: content/en/real_user_monitoring/application_monitoring/browser/setup/server/nginx.md

<div class="alert alert-danger">RUM Auto-Instrumentation is not available for the selected site ({{< region-param key="dd_site_name" >}}). Use <a href="/real_user_monitoring/application_monitoring/browser/setup/client">Client-Side instrumentation</a> instead.</div>
{{< /site-region >}}

## Overview

> RUM Auto-Instrumentation works by injecting the RUM Browser SDK into the HTML responses being served through a web server or proxy. This method uses the [NGINX Dynamic Modules capability][1] to implement a response body filter. The filter injects the RUM Browser SDK into the response body for responses identified as HTML. After auto-instrumentation is set up, you can manage configurations from the UI.

Reviewer: ```suggestion
RUM Auto-Instrumentation works by injecting the RUM Browser SDK into the HTML responses being served through a web server or proxy. This method uses the [NGINX Dynamic Modules capability][1] to implement a response body filter. The filter injects the RUM Browser SDK into the response body for responses identified as HTML. After auto-instrumentation is set up, you can manage configurations from the UI.
```
The link here resolves to `/tracing/other_telemetry/rum`, I was expecting it to resolve to:
https://docs.nginx.com/nginx/admin-guide/dynamic-modules/dynamic-modules/ as this was where it went to previously ?

{{< tabs >}}
{{% tab "Single-Step Instrumentation" %}}

Enables RUM Browser monitoring with [Single Step Instrumentation (SSI)][1].
When you run the Agent installation with RUM enabled, Datadog:
- Loads the NGINX module into your NGINX server through SSI
- Creates a RUM application for you
- Configures the NGINX module with the required RUM settings

**This approach requires no code changes and no manual NGINX configuration.**

1. Go to the [**Agent Installation**][2] page.
2. Select your platform (for example, Linux).
3. In the **Customize your observability coverage** section, enable **Real User Monitoring** under **Application Observability**.

   A RUM application is automatically created for you when you enable this option.

4. Copy the generated installation command and run it on your host.
5. Restart NGINX to begin collecting RUM sessions.

[1]: /tracing/trace_collection/single-step-apm/
[2]: https://app.datadoghq.com/fleet/install-agent/latest?platform=overview

{{% /tab %}}
{{% tab "Managed Instrumentation" %}}

Use this method if you need to configure sampling rates or application settings before deploying.

> 1. In Datadog, navigate to **Digital Experience > Manage Applications**, click [**New Application**][1], and select the JavaScript (JS) application type.

Reviewer: ```suggestion
1. In Datadog, navigate to **Digital Experience > Manage Applications**, click [**New Application**][1], and select the JavaScript (JS) application type.
```
On this page, I cannot find anything called "New Application" ?  Should it be click **Set up Manually**, then select JavaScript?

<img width="1258" height="507" alt="Screenshot 2026-06-18 at 3 58 22 PM" src="https://github.com/user-attachments/assets/a9179c2a-3e93-409f-8e9c-7d07ea82b7ec" />

2. Select **Auto-Instrumentation** and **NGINX**.
3. Configure your application parameters. See [guidance on configuring sampling][2].

> 4. Copy and run the installer command to load the Datadog NGINX module with the RUM SDK Injector onto NGINX.

Reviewer: ```suggestion
4. Copy and run the installer command to load the Datadog NGINX module with the RUM SDK Injector into your NGINX server.
```

   }
   ```

### Restart NGINX

> 1. Restart the NGINX server to begin collecting data for your Datadog RUM application. By default, the RUM SDK is injected to all HTML documents. You may need to clear your browser cache.

Reviewer: ```suggestion
1. Restart the NGINX server to begin collecting data for your Datadog RUM application. By default, the RUM SDK is injected into all HTML documents. You may need to clear your browser cache.
```
something claude flagged but I know this was here previously :)

2. (Optional) To verify the module is successfully injecting the RUM Browser SDK into HTML pages, check the NGINX error logs for relevant messages. The module logs important steps during the injection process. Confirm that NGINX is configured with at least the `INFO` log level with the following:

   ```javascript
   error_log <file> info;
   ```

{{% /tab %}}
{{< /tabs >}}

## Updating your RUM application

You can update your RUM application settings at any time. From the [Application Management][3] list, select your RUM application and navigate to the **SDK Configuration** page. Click **Save Changes** after making updates.

### Sampling rates

Adjust the slider or enter a specific percentage in the input box for Session Sampling or Session Replay Sampling. Copy and paste the configuration snippet to your `nginx.conf` file.

{{% rum-browser-auto-instrumentation-update-user-attributes %}}

## Troubleshooting

### NGINX stops responding

> If NGINX stops serving requests, specifically after installation, contact [Datadog support][4] with the following information:

Reviewer: ```suggestion
If NGINX stops serving requests after installation, contact [Datadog support][4] with the following information:
```

| 1.28.2 | [ngx_http_datadog-amd64-1.28.2][41] | [ngx_http_datadog-arm64-1.28.2][42] |
| 1.28.3 | [ngx_http_datadog-amd64-1.28.3][43] | [ngx_http_datadog-arm64-1.28.3][44] |
| 1.29.0 | [ngx_http_datadog-amd64-1.29.0][45] | [ngx_http_datadog-arm64-1.29.0][46] |
| 1.29.1 | [ngx_http_datadog-amd64-1.29.1][47] | [ngx_http_datadog-arm64-1.29.1][48] |
| 1.29.2 | [ngx_http_datadog-amd64-1.29.2][49] | [ngx_http_datadog-arm64-1.29.2][50] |
| 1.29.3 | [ngx_http_datadog-amd64-1.29.3][51] | [ngx_http_datadog-arm64-1.29.3][52] |
| 1.29.4 | [ngx_http_datadog-amd64-1.29.4][53] | [ngx_http_datadog-arm64-1.29.4][54] |
| 1.29.5 | [ngx_http_datadog-amd64-1.29.5][55] | [ngx_http_datadog-arm64-1.29.5][56] |
| 1.29.6 | [ngx_http_datadog-amd64-1.29.6][57] | [ngx_http_datadog-arm64-1.29.6][58] |
| 1.29.7 | [ngx_http_datadog-amd64-1.29.7][59] | [ngx_http_datadog-arm64-1.29.7][60] |
| 1.29.8 | [ngx_http_datadog-amd64-1.29.8][61] | [ngx_http_datadog-arm64-1.29.8][62] |
| 1.30.0 | [ngx_http_datadog-amd64-1.30.0][63] | [ngx_http_datadog-arm64-1.30.0][64] |

## Further reading

{{< partial name="whats-next/whats-next.html" >}}

> [1]: https://docs.nginx.com/nginx/admin-guide/dynamic-modules/dynamic-modules/

Reviewer: this link is resolving to `/tracing/other_telemetry/rum/` but appears correct here, maybe try merging in master ?

Author: completely changed the numbers. this should be resolved now. thanks for the find!
